pkg/master/master.go

```diff
@@ -179,16 +182,27 @@ func New(c *Config) (*Master, error) {
 		return nil, fmt.Errorf("Master.New() called with config.KubeletClient == nil")
 	}
 
-	s, err := c.Config.New()
+	gc := *c.Config                                              // copy before mutations
+	gc.EnableSwaggerUI = gc.EnableSwaggerUI && c.EnableUISupport // disable swagger UI if general UI supports is
+	s, err := gc.New()
 	if err != nil {
 		return nil, err
 	}
 
+	if c.EnableUISupport {
+		routes.UIRedirect{}.Install(s.Mux, s.HandlerContainer)
+	}
+	if c.EnableLogsSupport {
+		routes.Logs{}.Install(s.Mux, s.HandlerContainer)
+	}
+
 	m := &Master{
 		GenericAPIServer:        s,
 		enableCoreControllers:   c.EnableCoreControllers,
 		deleteCollectionWorkers: c.DeleteCollectionWorkers,
 		tunneler:                c.Tunneler,
+		enableUISupport:         c.EnableUISupport,
```

Comment:
Good point. There were some tests for them. That's the only use. Will remove them.